### DISCUSSION TITLE
How to create an Hex that can be used in JS UI with Extrinsic / Decode?

### DISCUSSION THREAD

**Discussion by johnuopini** [polkascan/py-substrate-interface#374] [Q&A] [answered]
Is there a way to encode the compose_call output in a way that can be understood in JS UI using Extrinsic -> Decode? I cant find an equivalent to toHex() in JS.

**Comment by arjanz** [comment, marked as answer]
You can output the hex of the encoded SCALE data using `call.data.to_hex()`:

```python
call = substrate.compose_call(
    call_module='Balances',
    call_function='transfer',
    call_params={
        'dest': '5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty',
        'value': 1 * 10**15
    }
)

print(call.data.to_hex())
```

For the other way around, see: https://github.com/polkascan/py-substrate-interface/discussions/361

**Comment by johnuopini** [comment]
Works like a charm, thanks!
